docusaurus/docs/iOS/uikit/guides/working-with-attachments.md

```diff
@@ -321,4 +321,151 @@ class WorkoutAttachmentViewInjector: AttachmentViewInjector {
 Finally, don't forget to assign the custom message list if you didn't yet:
 ```swift
 Components.default.messageListVC = CustomChatMessageListVC.self
-```
\ No newline at end of file
+```
+
+### Tracking custom attachment upload progress
+In the previous examples, we assumed that you already have the custom attachment remote URL available. But, you can also upload the custom attachment through the Stream's SDK and observe the uploading progress.
+
+The simplest approach is to upload through the `channelController.uploadAttachment()`. Using this function you can observe the progress, but the limitation is that at this time, the message is not yet sent, so this approach is good if you want to show the upload progress before creating the message. Here is an example:
+
+```swift
+let controller = client.channelController(for: ChannelId(type: .messaging, id: "my-test-channel"))
```

Comment:
I think it is assumed you will do this in your custom UI, it was already like this before, in the previous examples, so the customer has already some context. When we will have a full example in our DemoApp, then it will be even more clear.